### ISSUE TITLE
services providers need to take delegations they can use

### ISSUE THREAD

**Issue by Gozala** [web3-storage/w3protocol#265]
There is some background story here https://github.com/web3-storage/specs/pull/7, but general idea is that we want to have a chain of command

```mermaid
flowchart TB
   root>"📜 did:web:web3.storage#top"]-->| rotate on compromise  |manager("🔑 did:web:web3.storage#manager")

manager-->|rotate regularly|access("🤖 did:web:web3.storage#access")
manager-->|rotate regularly|upload("🤖 did:web:web3.storage#upload")
```

Where top key is in the safe deposit box, manager key(s) are only used to delegate to individual workers. This implies that our services will only have worker keys, however they may need to act on behalf of the root meaning they should be able to issue following capability to the user when they authenticate

```ts
{ with: 'did:web:web3.storage', can: './update' }
```

In other words services need a to be given authority to do so in form of delegation chain which they can utilize

```ts
{
  iss: "did:key:rotate",
  aud: "did:key:top",
  att: [{ with: 'did:web:web3.storage', can: "*" }],
  prf: [{
    iss: 'did:key:top',
    with: 'did:web:web3.storage',
    att: [{ with: "did:web:web3.storage", can: "*" }],
    prf: [{
      iss: 'did:web:web3.storage',
      aud: 'did:key:top',
      att: [{ with: "did:web:web3.storage", can: "*" }]
    }]
  }]
}
```

So they can utilize it when performing delegations like those. These will also become necessary for invocation receipts

**Comment by Gozala** [comment]
In the short term we can just put root keys into workers, but we should plan to do something along the lines of above down the line.

**Comment by Gozala** [comment]
I've updated example such that only root DID is did:web and rest are did:key that is because use of did:web in other positions implies more lookups or coordination for no obvious benefit, which is why I think in those cases we should use did:key's instead. That's not to say we don't publish them in DID document, we probably should. Even then we should probably refer to them in a form that does not requires resolution which is did:key
